zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -261,6 +261,12 @@ public int getNumIdleClient() {
     }
   }
 
+  public void setMaxPoolSize(int size) {
+    if (clientPool != null) {
+      //Size + 2 for progress poller , cancel operation
+      clientPool.setMaxTotal(size + 2);
```

Comment:
I suggest not limit the thrift client pool size. RemoteInterpreter's maximum concurrency is already limited (1 on FIFO,  maxPoolSize on Parallel scheduler), so i think it'll generally safe without limiting thrift clientpool size.

Also limiting two different concurrency parameters (one concurrency of job in scheduler, the other number of thrift client connection in the pool) with a single value (maxPoolSize) is not the finest way i think.